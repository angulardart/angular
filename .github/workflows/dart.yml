# Created with package:mono_repo v5.0.2
name: Dart CI
on:
  push:
    branches:
      - master
      - google
  pull_request:
    branches:
      - master
      - google
defaults:
  run:
    shell: bash
env:
  PUB_ENVIRONMENT: bot.github

jobs:
  job_001:
    name: mono_repo self validate
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:stable"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: stable
      - id: checkout
        uses: actions/checkout@v2.3.4
      - name: mono_repo self validate
        run: dart pub global activate mono_repo 5.0.2
      - name: mono_repo self validate
        run: dart pub global run mono_repo generate --validate
  job_002:
    name: "analyze; PKG: angular_ast; `dart analyze`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:analyze_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: angular_ast; dart analyze
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: dart analyze
  job_003:
    name: "analyze; PKGS: angular_compiler, examples/hello_world; `dart analyze --fatal-infos`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world;commands:analyze_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler-examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; dart analyze --fatal-infos"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart analyze --fatal-infos
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; dart analyze --fatal-infos"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart analyze --fatal-infos
  job_004:
    name: "build; PKG: _tests; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner build --fail-on-severe"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_005:
    name: "build; PKG: angular_ast; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_006:
    name: "build; PKG: angular_compiler; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_007:
    name: "build; PKG: angular_forms; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_008:
    name: "build; PKG: angular_router; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_009:
    name: "build; PKG: angular_test; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner build --fail-on-severe"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_010:
    name: "build; PKG: examples/hello_world; `pub run build_runner build --fail-on-severe`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world;commands:command_0"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:examples/hello_world
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: examples_hello_world_pub_upgrade
        name: examples/hello_world; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: examples/hello_world
        run: dart pub upgrade
      - name: "examples/hello_world; pub run build_runner build --fail-on-severe"
        if: "always() && steps.examples_hello_world_pub_upgrade.conclusion == 'success'"
        working-directory: examples/hello_world
        run: pub run build_runner build --fail-on-severe
    needs:
      - job_001
      - job_002
      - job_003
  job_011:
    name: "unit_test; PKG: _tests; `pub run build_runner test --fail-on-severe -- -P browser`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_2"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run build_runner test --fail-on-severe -- -P browser"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run build_runner test --fail-on-severe -- -P browser
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_012:
    name: "unit_test; PKG: _tests; `pub run test -P vm`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests;commands:command_1"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:_tests
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: _tests_pub_upgrade
        name: _tests; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: _tests
        run: dart pub upgrade
      - name: "_tests; pub run test -P vm"
        if: "always() && steps._tests_pub_upgrade.conclusion == 'success'"
        working-directory: _tests
        run: pub run test -P vm
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_013:
    name: "unit_test; PKG: angular_ast; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_ast
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_ast_pub_upgrade
        name: angular_ast; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_ast
        run: dart pub upgrade
      - name: "angular_ast; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_ast_pub_upgrade.conclusion == 'success'"
        working-directory: angular_ast
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_014:
    name: "unit_test; PKG: angular_compiler; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_compiler
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_compiler_pub_upgrade
        name: angular_compiler; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_compiler
        run: dart pub upgrade
      - name: "angular_compiler; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_compiler_pub_upgrade.conclusion == 'success'"
        working-directory: angular_compiler
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_015:
    name: "unit_test; PKG: angular_forms; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_forms
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_forms_pub_upgrade
        name: angular_forms; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_forms
        run: dart pub upgrade
      - name: "angular_forms; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_forms_pub_upgrade.conclusion == 'success'"
        working-directory: angular_forms
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_016:
    name: "unit_test; PKG: angular_router; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_router
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_router_pub_upgrade
        name: angular_router; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_router
        run: dart pub upgrade
      - name: "angular_router; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_router_pub_upgrade.conclusion == 'success'"
        working-directory: angular_router
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_017:
    name: "unit_test; PKG: angular_test; `pub run build_runner test --fail-on-severe -- -P ci`"
    runs-on: ubuntu-latest
    steps:
      - name: Cache Pub hosted dependencies
        uses: actions/cache@v2.1.6
        with:
          path: "~/.pub-cache/hosted"
          key: "os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test;commands:command_3"
          restore-keys: |
            os:ubuntu-latest;pub-cache-hosted;dart:dev;packages:angular_test
            os:ubuntu-latest;pub-cache-hosted;dart:dev
            os:ubuntu-latest;pub-cache-hosted
            os:ubuntu-latest
      - uses: dart-lang/setup-dart@v1.2
        with:
          sdk: dev
      - id: checkout
        uses: actions/checkout@v2.3.4
      - id: angular_test_pub_upgrade
        name: angular_test; dart pub upgrade
        if: "always() && steps.checkout.conclusion == 'success'"
        working-directory: angular_test
        run: dart pub upgrade
      - name: "angular_test; pub run build_runner test --fail-on-severe -- -P ci"
        if: "always() && steps.angular_test_pub_upgrade.conclusion == 'success'"
        working-directory: angular_test
        run: pub run build_runner test --fail-on-severe -- -P ci
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
  job_018:
    name: Notify failure
    if: "(github.event_name == 'push' || github.event_name == 'schedule') && failure()"
    runs-on: ubuntu-latest
    steps:
      - run: |
          curl -H "Content-Type: application/json" -X POST -d \
            "{'text':'Build failed! ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}'}" \
            "${CHAT_WEBHOOK_URL}"
        env:
          CHAT_WEBHOOK_URL: "${{ secrets.TEAM_CHAT_WEBHOOK_URL }}"
    needs:
      - job_001
      - job_002
      - job_003
      - job_004
      - job_005
      - job_006
      - job_007
      - job_008
      - job_009
      - job_010
      - job_011
      - job_012
      - job_013
      - job_014
      - job_015
      - job_016
      - job_017
