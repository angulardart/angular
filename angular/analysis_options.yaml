# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
