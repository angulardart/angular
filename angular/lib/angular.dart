/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
