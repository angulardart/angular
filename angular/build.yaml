targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
