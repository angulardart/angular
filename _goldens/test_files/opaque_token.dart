import 'package:angular/angular.dart';

const baseUrlToken = const OpaqueToken('baseUrlDescription');

@Component(
  selector: 'has-opaque-tokens',
  template: '{{baseUrl}}',
  providers: const [
    const Provider(baseUrlToken, useValue: 'https://localhost'),
  ],
  // TODO(b/71710685): Change to `Visibility.local` to reduce code size.
  visibility: Visibility.all,
)
class HasOpaqueTokens {
  final String baseUrl;

  HasOpaqueTokens(@Inject(baseUrlToken) this.baseUrl);
}

const listOfDurationToken = const OpaqueToken<List<Duration>>('listOfDuration');

@Component(
  selector: 'contains-child-component',
  template: r'''
    <div *ngIf="someValue">
      <div *ngIf="someValue">
        <injects-typed-token-from-parent></injects-typed-token-from-parent>
      </div>
    </div>
  ''',
  directives: const [
    InjectsTypedTokenFromSomeParent,
    NgIf,
  ],
  visibility: Visibility.local,
)
class ContainsChildComponent {
  bool someValue = true;
}

@Component(
  selector: 'injects-typed-token-from-parent',
  template: '',
  visibility: Visibility.local,
)
class InjectsTypedTokenFromSomeParent {
  final List<Duration> example;

  InjectsTypedTokenFromSomeParent(@Inject(listOfDurationToken) this.example);
}
