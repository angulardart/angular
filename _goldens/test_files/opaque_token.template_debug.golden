// GENERATED CODE - DO NOT MODIFY BY HAND

// **************************************************************************
// Generator: TemplateGenerator
// **************************************************************************

// ignore_for_file: cancel_subscriptions,constant_identifier_names,duplicate_import,non_constant_identifier_names,library_prefixes,UNUSED_IMPORT,UNUSED_SHOWN_NAME
import 'opaque_token.dart';
export 'opaque_token.dart';
import 'package:angular/angular.dart';
// Required for initReflector().
import 'package:angular/src/di/reflector.dart'
    as _ngRef;
import 'package:angular/angular.template.dart'
    as _ref0;

import 'package:angular/src/debug/debug_context.dart';
import 'package:angular/src/debug/debug_app_view.dart';
import 'opaque_token.dart' as import2;
import 'dart:html' as import3;
import 'package:angular/src/core/render/api.dart';
import 'package:angular/src/core/linker/app_view.dart';
import 'package:angular/src/core/linker/view_type.dart'
    as import6;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'package:angular/src/core/linker/app_view_utils.dart'
    as import8;
import 'package:angular/angular.dart';

const List<dynamic> styles$HasOpaqueTokens = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_HasOpaqueTokens0 = [null];

class ViewHasOpaqueTokens0 extends DebugAppView<import2.HasOpaqueTokens> {
  import3.Text _text_0;
  static RenderComponentType _renderType;
  ViewHasOpaqueTokens0(AppView<dynamic> parentView, num parentIndex)
      : super(
            import6.ViewType.COMPONENT,
            {},
            parentView,
            parentIndex,
            ChangeDetectionStrategy.CheckAlways,
            nodeDebugInfos_HasOpaqueTokens0) {
    rootEl = import3.document.createElement('has-opaque-tokens');
    _renderType ??= import8.appViewUtils.createRenderType(
        'asset:_goldens/test_files/opaque_token.dart class HasOpaqueTokens - inline template',
        ViewEncapsulation.None,
        styles$HasOpaqueTokens);
    setupComponentType(_renderType);
  }
  @override
  ComponentRef<import2.HasOpaqueTokens> build() {
    final import3.HtmlElement parentRenderNode = initViewRoot(rootEl);
    _text_0 = new import3.Text('');
    parentRenderNode.append(_text_0);
    dbgElm(this, _text_0, 0, 0, 0);
    init(const [], null, [_text_0]);
    return null;
  }

  @override
  void detectChangesInternal() {
    final import2.HasOpaqueTokens _ctx = ctx;
    bool firstCheck = (this.cdState == 0);
    if (firstCheck) {
      (_text_0.text = (_ctx.baseUrl ?? ''));
    }
  }
}

AppView<import2.HasOpaqueTokens> viewFactory_HasOpaqueTokens0(
    AppView<dynamic> parentView, num parentIndex) {
  return new ViewHasOpaqueTokens0(parentView, parentIndex);
}

const List<dynamic> styles$HasOpaqueTokensHost = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_HasOpaqueTokensHost0 = [
  new StaticNodeDebugInfo(
      [
        const OpaqueToken<dynamic>('baseUrlDescription'),
        import2.HasOpaqueTokens
      ],
      import2.HasOpaqueTokens,
      <String, dynamic>{})
];

class _ViewHasOpaqueTokensHost0 extends DebugAppView<dynamic> {
  ViewHasOpaqueTokens0 _compView_0;
  String _const_OpaqueToken_dynamic___baseUrlDescription___0_4;
  import2.HasOpaqueTokens _HasOpaqueTokens_0_5;
  _ViewHasOpaqueTokensHost0(AppView<dynamic> parentView, num parentIndex)
      : super(
            import6.ViewType.HOST,
            {},
            parentView,
            parentIndex,
            ChangeDetectionStrategy.CheckAlways,
            nodeDebugInfos_HasOpaqueTokensHost0);
  @override
  ComponentRef build() {
    _compView_0 = new ViewHasOpaqueTokens0(this, 0);
    rootEl = _compView_0.rootEl;
    dbgIdx(rootEl, 0);
    _const_OpaqueToken_dynamic___baseUrlDescription___0_4 = 'https://localhost';
    _HasOpaqueTokens_0_5 = new import2.HasOpaqueTokens(
        _const_OpaqueToken_dynamic___baseUrlDescription___0_4);
    _compView_0.create(_HasOpaqueTokens_0_5, projectableNodes);
    init0Dbg(rootEl, [rootEl]);
    return new ComponentRef<import2.HasOpaqueTokens>(
        0, this, rootEl, _HasOpaqueTokens_0_5);
  }

  @override
  dynamic injectorGetInternal(
      dynamic token, int nodeIndex, dynamic notFoundResult) {
    if ((identical(token, const OpaqueToken<dynamic>('baseUrlDescription')) &&
        (0 == nodeIndex))) {
      return _const_OpaqueToken_dynamic___baseUrlDescription___0_4;
    }
    if ((identical(token, import2.HasOpaqueTokens) && (0 == nodeIndex))) {
      return _HasOpaqueTokens_0_5;
    }
    return notFoundResult;
  }

  @override
  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  @override
  void destroyInternal() {
    _compView_0?.destroy();
  }
}

AppView viewFactory_HasOpaqueTokensHost0(
    AppView<dynamic> parentView, num parentIndex) {
  return new _ViewHasOpaqueTokensHost0(parentView, parentIndex);
}

const ComponentFactory<import2.HasOpaqueTokens> HasOpaqueTokensNgFactory =
    const ComponentFactory<import2.HasOpaqueTokens>('has-opaque-tokens',
        viewFactory_HasOpaqueTokensHost0, _HasOpaqueTokensMetadata);
const _HasOpaqueTokensMetadata = const [];
var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;
  _ref0.initReflector();
  _ngRef.registerComponent(
    HasOpaqueTokens,
    HasOpaqueTokensNgFactory,
  );
}
